### PR TITLE
Cleaned Experiment/views.py

### PR DESCRIPTION
**Pull request by meshde**
Cleared all references to Board,Slot and Booking models in experiment/views.py. Also changed the Experiment model schema to include Foreign Key reference to Account.(Previously, Experiment had foreign key to Booking and Booking in turn had foreign key to Account)

The client is successfully able to connect to the modified local django server on my PC with the message displayed as "Ready to execute Scilab code"

### CODE REVIEW

**Comment by meshde** [comment]
I have also completely changed the sbhs.py file, the getTemp() method now returns the sum of the fan and heat percentages.
The code is working with the scilab experiments and correct output is being displayed in the client application window

I have just commented out the previous codes rather than deleting them.This is to facilitate quick debugging and fixing, I'll be sure to completely clean the code once all possible changes have been made and the code works perfectly

File: experiments/meshde/2017Feb24_17_32_51.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+0 0 100 100.00 1487937823278 1487937823288 1487937823350 0.10000E+01
+1 30 50 80.00 1487937841792 1487937841797 1487937841850 0.10000E+01
+2 30 50 80.00 1487937848450 1487937848457 1487937848492 0.20000E+01
+3 30 50 80.00 1487937848676 1487937848684 1487937848720 0.30000E+01
+4 30 50 80.00 1487937848847 1487937848859 1487937848896 0.40000E+01
+5 30 50 80.00 1487937849021 1487937849028 1487937849064 0.50000E+01
+6 30 50 80.00 1487937849189 1487937849199 1487937849239 0.60000E+01
```
Comment (maheshgudi) [inline experiments/meshde/2017Feb24_17_32_51.txt:7]: Please get rid of these files. They are not useful to a developer.

File: sbhs_server/experiment/views.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 from django.contrib.auth.decorators import login_required
 from django.contrib.auth import authenticate
 from django.contrib.auth import login as LOGIN
-from sbhs_server.tables.models import Slot, Account, Experiment, Booking
+from sbhs_server.tables.models import Account, Experiment
 import json, datetime, os, time
```
Comment (maheshgudi) [inline sbhs_server/experiment/views.py]: Import the modules individually.

File: sbhs_server/experiment/views.py [hunk 3]
```diff
@@ -18,61 +18,33 @@ def initiation(req):
...
+            f.close()
+            
+
+            LOGIN(req, user)
+            e = Experiment()
+            e.user = user
+            e.log=os.path.join(logdir, filename)
+            e.save()
+
+            #key = str(1)#Temporarily till SBHS class and settings.board dictionary are changed
```
Comment (maheshgudi) [inline sbhs_server/experiment/views.py]: Get rid of the commented code

File: sbhs_server/tables/migrations/0008_initial.py [hunk 4]
```diff
@@ -8,17 +8,6 @@
 class Migration(SchemaMigration):
 
     def forwards(self, orm):
```
Comment (maheshgudi) [inline sbhs_server/tables/migrations/0008_initial.py]: Get rid of the migration files. They are not needed
 Add them in gitignore

**Comment by meshde** [comment]
I've removed each and every reference to Slot,Webcam,Bookings, and board.
The site and the client experiment are working perfectly
I've also added the suggested directories in the gitignore, but they still have been pushed. Could you please check if I did anything wrong with respect to the gitignore file?

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,5 @@
+*.pyc
+experiments\
+log\
+*.sqlite3
+migrations\
```
Comment (maheshgudi) [inline .gitignore]: add a new line here. All files should end on a new line.

File: sbhs_server/admin/views.py [hunk 6]
```diff
@@ -1,7 +1,7 @@
 from django.shortcuts import render, redirect
 from django.http import Http404
 from django.contrib.auth.decorators import login_required
-from sbhs_server.tables.models import Board, Booking
+#from sbhs_server.tables.models import Board, Booking
```
Comment (maheshgudi) [inline sbhs_server/admin/views.py]: commented code.

File: sbhs_server/admin/views.py [hunk 7]
```diff
@@ -10,30 +10,30 @@ def checkadmin(req):
     if not req.user.is_admin:
         raise Http404
 
-@login_required(redirect_field_name=None)
-def index(req):
-    checkadmin(req)
-    boards = Board.objects.order_by('online').all()
-    allotment_mode = "Random" if Board.can_do_random_allotment() else "Workshop"
-    return render(req, 'admin/index.html', {"boards": boards, "allotment_mode": allotment_mode})
+#@login_required(redirect_field_name=None)
```
Comment (maheshgudi) [inline sbhs_server/admin/views.py]: Commented code.

File: sbhs_server/experiment/views.py [hunk 8]
```diff
@@ -3,76 +3,47 @@
 from django.contrib.auth.decorators import login_required
 from django.contrib.auth import authenticate
 from django.contrib.auth import login as LOGIN
-from sbhs_server.tables.models import Slot, Account, Experiment, Booking
+from sbhs_server.tables.models import Account, Experiment
 import json, datetime, os, time
```
Comment (maheshgudi) [inline sbhs_server/experiment/views.py]: Please import the modules separately as asked.

File: sbhs_server/experiment/views.py [hunk 9]
```diff
@@ -3,76 +3,47 @@
...
 from django.views.decorators.csrf import csrf_exempt
-from sbhs_server import settings
+from sbhs_server import settings,sbhs
 # Create your views here.
 # 
+
+global boards
```
Comment (maheshgudi) [inline sbhs_server/experiment/views.py]: What is boards? Why are you putting that in global namespace?

File: sbhs_server/experiment/views.py [hunk 10]
```diff
@@ -3,76 +3,47 @@
...
 def check_connection(req):
     return HttpResponse("TESTOK")
 
 @csrf_exempt
 def initiation(req):
+    global boards
```
Comment (maheshgudi) [inline sbhs_server/experiment/views.py]: Again? Why this?

File: sbhs_server/experiment/views.py [hunk 11]
```diff
@@ -87,77 +58,67 @@ def initiation(req):
...
-                STATUS = 0
-                MESSAGE = "Slot has ended. Please book the next slot to continue the experiment."
-
-                reset(req)
-                boards[key]["experiment_id"] = None
-        else:
-            STATUS = 0
-            MESSAGE = "You haven't booked this slot."
-
+        global boards
```
Comment (maheshgudi) [inline sbhs_server/experiment/views.py]: Why?

File: sbhs_server/experiment/views.py [hunk 12]
```diff
@@ -87,77 +58,67 @@ def initiation(req):
...
-                boards[key]["board"].setFan(100)
-                log_data(boards[key]["board"], key)
-                if endtime < now:
-                    boards[key]["experiment_id"] = None
+			
+            experiment = Experiment.objects.select_related().filter(user_id=user.id).order_by("-id")
+
+            experiment = experiment[0]
+            now = datetime.datetime.now()
+            #boards.setHeat(0)
```
Comment (maheshgudi) [inline sbhs_server/experiment/views.py]: commented code

File: sbhs_server/settings.py [hunk 13]
```diff
@@ -203,8 +201,8 @@
     }
 
 EXPERIMENT_LOGS_DIR = os.path.join(BASE_DIR, 'experiments')
-WEBCAM_DIR = os.path.join(STATIC_ROOT, 'img/webcam/') if is_production else os.path.join(BASE_DIR, 'static/img/webcam/')
-WEBCAM_STATIC_DIR = 'img/webcam/'
+#WEBCAM_DIR = os.path.join(STATIC_ROOT, 'img/webcam/') if is_production else os.path.join(BASE_DIR, 'static/img/webcam/')
```
Comment (maheshgudi) [inline sbhs_server/settings.py]: get rid of the code.

**Review by maheshgudi — CHANGES_REQUESTED** [review]
Need to clean up the code further.

File: .gitignore (hunk 5 above)
Comment (meshde) [reply .gitignore]: You mean a blank last line?

File: sbhs_server/experiment/views.py (hunk 8 above)
Comment (meshde) [reply sbhs_server/experiment/views.py]: I'm sorry, forgot about these.I'll change them right away

File: sbhs_server/experiment/views.py (hunk 9 above)
Comment (meshde) [reply sbhs_server/experiment/views.py]: 'boards' is a Sbhs object that is created when the user logs in from the client application in the initiation() method. I've declared it global as the object created during initiation() is being used in the experiment() method.

File: .gitignore (hunk 5 above)
Comment (maheshgudi) [reply .gitignore]: yes

**Comment by meshde** [comment]
I've made all changes that I was asked to make. The things that remain to be done include changing the admin views to display some relevant info(I haven't been told what to do about it) and integrating with yaksh module.

**Comment by rupakrokade** [comment]
Merged mehmoods work with the base repo

**Pull request merged**